cmd/lakectl/cmd/annotate.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+const annotateTemplate = `{{  $val := .Commit}}
+{{ $.Object|ljust 15}} {{ $val.Committer|ljust 20 }} {{ $val.Id | printf "%.16s"|ljust 20 }} {{ $val.CreationDate|date }}  {{ $.CommitMessage |ljust 30 }}
+`
+
+var annotateCmd = &cobra.Command{
+	Use:   "annotate <path uri>",
+	Short: "List entries under a given path, annotating each with the latest modifying commit ",
+	// Long:    "Show who created the latest commit to an object or objects in a give path",
+	Aliases: []string{"blame"},
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		pathURI := MustParsePathURI("path", args[0])
+		client := getClient()
+		amount := 1
+		pfx := api.PaginationPrefix(*pathURI.Path)
+		res, err := client.ListObjectsWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.ListObjectsParams{Prefix: &pfx})
+		DieOnResponseError(res, err)
+		var from string
+		for {
+			pfx := api.PaginationPrefix(*pathURI.Path)
+			params := &api.ListObjectsParams{
+				Prefix: &pfx,
+				After:  api.PaginationAfterPtr(from),
+			}
+			resp, err := client.ListObjectsWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, params)
+			DieOnResponseError(resp, err)
+			for _, obj := range resp.JSON200.Results {
+				prfx := []string{obj.Path}
+				logCommitsParams := &api.LogCommitsParams{
+					Amount:  api.PaginationAmountPtr(amount),
+					Objects: &prfx,
```

Comment:
api.StringPtr

Reply:
Not sure what you mean here.

Reply:
I revert my comment - I thought this is a string and you pass a pointer to the API - we have a function for that in 'api' package. But this is not the case - so, ignore.